src/SmartEntity/SmartEntity.php

```diff
@@ -0,0 +1,198 @@
+     */
+    private function isInternalType($type): bool
+    {
+        return is_numeric($type) && array_key_exists($type, self::INTERNAL_TYPES);
+    }
+
+    private function isPropertySet($propertyName): bool
+    {
+        return array_key_exists($propertyName, $this->properties);
+    }
+
+    /**
+     * @param mixed $value
+     * @param string $class
+     * @return mixed
+     * @throws Exception
+     * @throws InvalidArgumentException
+     */
+    private function processObjectType($value, string $class)
+    {
+        $this->isClassOrInterface($class);
+
+        if (is_array($value) && is_subclass_of($class, Makeable::class)) {
+            return $class::make($value);
+        }
+
+        if (is_a($value, $class)) {
+            return $value;
+        }
+
+        throw new InvalidArgumentException(
+            'Expected value to be object of [' . $class . '] type different type was given'
+        );
+    }
+
+    /**
+     * @param string $class
+     * @return bool
+     * @throws Exception
+     */
+    private function isClassOrInterface(string $class): bool
+    {
+        if (class_exists($class) || interface_exists($class)) {
+            return true;
+        }
+
+        throw new Exception('Non existing class or interface [' . $class . ']');
+    }
+
+    private function isTypeArrayOfObjects($type): bool
+    {
+        return strpos($type, '[]') !== false;
+    }
+
+    /**
+     * @param array $value
+     * @param mixed $type
+     * @return array
+     * @throws Exception
+     * @throws InvalidArgumentException
+     */
+    private function processArrayOfObj(array $value, $type): array
+    {
+        $ar = [];
+        $class = $this->getClassNameFromType($type);
+
+        foreach ($value as $key => $obj) {
+            $ar[$key] = $this->processObjectType($obj, $class);
+        }
+
+        return $ar;
+    }
+
+    private function getClassNameFromType($type): string
+    {
+        return (string) str_replace(self::EXT_ARRAY, '', $type);
+    }
+
+    private function validateInternalType($value, $type): bool
+    {
+        switch ($type) {
+            case self::TYPE_ARRAY:
+                return is_array($value);
+            case self::TYPE_INT:
+                return is_int($value);
+            case self::TYPE_STRING:
+                return is_string($value);
+            case self::TYPE_BOOL:
+                return is_bool($value);
+            case self::TYPE_FLOAT:
+                return is_float($value);
+            case self::TYPE_NUMERIC:
+                return is_numeric($value);
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $property
+     * @param array $arguments
+     * @return $this
+     * @throws Exception
+     * @throws InvalidArgumentException
+     */
+    protected function add(string $property, array $arguments)
+    {
+        $this->fetchValue($arguments, $property);
+
+        if (!array_key_exists($property, $this->properties) ||
```

Comment:
```suggestion
        if (
           !array_key_exists($property, $this->properties) ||
```
https://www.php-fig.org/psr/psr-12/#51-if-elseif-else

Reply:
- not sure if we decided to follow PSR-12 yet, but it's fine